File: java/src/main/java/de/gdata/vaas/messages/Detection.java
package de.gdata.vaas.messages;

import com.google.gson.annotations.SerializedName;

import lombok.Getter;
import lombok.NonNull;

@Getter
public class Detection {
    int engine;

    @NonNull
    @SerializedName("file_name")
    String fileName;

    @NonNull
    String virus;
}

File: dotnet/Vaas/src/Vaas/Messages/Detection.cs
using System.Text.Json.Serialization;

namespace Vaas.Messages;

public class Detection
{
    [JsonPropertyName("engine")]
    public int? Engine { get; init; }

    [JsonPropertyName("file_name")]
    public string FileName { get; init; }

    [JsonPropertyName("virus")]
    public string Virus { get; init; }
}